src/d3a/models/area/markets.py

```diff
@@ -101,37 +175,52 @@ def select_market_class(is_spot_market):
         else:
             return BalancingMarket
 
-    def create_future_markets(self, current_time, is_spot_market, area):
+    def create_future_markets(self, current_time: DateTime,
+                              is_spot_market: bool, area: "Area") -> bool:
+        """Create future markets according to the market count."""
         markets = self.markets if is_spot_market else self.balancing_markets
-        market_class = self.select_market_class(is_spot_market)
+        market_class = self._select_market_class(is_spot_market)
 
         changed = False
         for offset in (area.config.slot_length * i
                        for i in range(area.config.market_count)):
-            timeframe = current_time + offset
-            if timeframe not in markets:
-                # Create markets for missing slots
-                market = market_class(
-                    time_slot=timeframe,
-                    bc=area.bc,
-                    notification_listener=area.dispatcher.broadcast_callback,
-                    grid_fee_type=area.config.grid_fee_type,
-                    grid_fees=GridFee(grid_fee_percentage=area.grid_fee_percentage,
-                                      grid_fee_const=area.grid_fee_constant),
-                    name=area.name,
-                    in_sim_duration=is_timeslot_in_simulation_duration(area.config, timeframe)
-                )
-
-                area.dispatcher.create_area_agents(is_spot_market, market)
-                markets[timeframe] = market
+            time_slot = current_time + offset
+            if time_slot not in markets:
+                markets[time_slot] = self._create_market(market_class, time_slot, area,
+                                                         is_spot_market)
                 changed = True
                 self.log.trace("Adding {t:{format}} market".format(
-                    t=timeframe,
-                    format="%H:%M"
-                           if area.config.slot_length.seconds > 60
-                    else "%H:%M:%S"
+                    t=time_slot,
+                    format="%H:%M" if area.config.slot_length.seconds > 60 else "%H:%M:%S"
                 ))
-        self._indexed_future_markets = {
-            m.id: m for m in self.all_future_spot_markets
-        }
+        self._update_indexed_future_markets()
         return changed
+
+    def create_settlement_market(self, time_slot: DateTime, area: "Area") -> None:
+        """Create a new settlement market."""
+        self.settlement_markets[time_slot] = (
+            self._create_market(market_class=self._select_market_class(is_spot_market=True),
```

Comment:
I have changed it in my PR, however IMO the settlement market is always double-sided by definition